### PR TITLE
Feature/improve te test

### PR DESCRIPTION
**Pull request by kpanier**
New TE Testcase to create a new Project.
Improvments in the fixture and swt aml
Support to create a project as testdate without ui

### CODE REVIEW

File: rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java [hunk 1]
```diff
@@ -110,9 +131,109 @@ public void wait(String seconds) throws NumberFormatException, InterruptedExcept
...
+	 *            of the tree.
+	 * @param itemName
+	 *            path to the item to be selectd. Example:
+	 *            "Root/SubNode/FinalNode"
+	 */
+	@FixtureMethod
+	public void selectElementInTreeView(String locator, String itemName) {
+		logger.trace("search for view with titel: {}", locator);
+		SWTBotTree tree = null;
+		if (locator.startsWith("[Single]")) {
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java]: This is not explained in the JavaDoc, what is the semantic?

File: rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java [hunk 2]
```diff
@@ -110,9 +131,109 @@ public void wait(String seconds) throws NumberFormatException, InterruptedExcept
...
+	 */
+	@FixtureMethod
+	public void executeContextMenuEntry(String viewName, String menuItem) throws Exception {
+		logger.trace("search for view with titel: {}", viewName);
+		SWTBotView view = bot.viewByTitle(viewName);
+		assertNotNull(view);
+		SWTBotTree tree = view.bot().tree();
+		assertNotNull(tree);
+		MenuItem item = ContextMenuHelper.contextMenu(tree, menuItem.split("/"));
+		new SWTBotMenu(item).click();
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java]: Shouldn't we check if the item is null here as well?

File: rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java [hunk 3]
```diff
@@ -110,9 +131,109 @@ public void wait(String seconds) throws NumberFormatException, InterruptedExcept
...
+		logger.trace("Yeaa renne weiter");
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java]: ?! ;)

File: rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java [hunk 4]
```diff
@@ -110,9 +131,109 @@ public void wait(String seconds) throws NumberFormatException, InterruptedExcept
...
+	 * @param value
+	 *            set to the widget.
+	 */
+	@FixtureMethod
+	public void typeInto(String locator, String value) {
+		logger.trace("search for text with titel: {}", locator);
+		SWTBotText text = null;
+		if (locator.startsWith("[Single]")) {
+			text = bot.text();
+		}
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java]: What happens if the locator does not start with `[Single]`? Shouldn't the locator be passed to the swt bot?

File: rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java [hunk 5]
```diff
@@ -110,9 +131,109 @@ public void wait(String seconds) throws NumberFormatException, InterruptedExcept
...
+	 *            to identify the button.
+	 */
+	@FixtureMethod
+	public void clickOn(String locator) {
+		logger.trace("search for button with titel: {}", locator);
+		SWTBotButton button = null;
+		if (locator.startsWith("[Label]")) {
+			String locatorFragment = getLocatorFragmentFrom(locator);
+			logger.trace("Fragment>>>>>>>>>>>>>>>>>> {}", locatorFragment);
+			this.reportWidgets();
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java]: Those two lines above look like debugging leftovers.

File: rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java (hunk 1 above)
Comment (kpanier) [inline rcp/org.testeditor.rcp4.uatests/src/main/java/org/testeditor/fixture/swt/SWTFixture.java]: We need a discussion about locator types. That is adressed in https://testeditor.atlassian.net/browse/TE-350.
In this implemenation are some examples about the need for that. In the current documentation it is hieded, until we are clear about the concept.

**Pull request merged**